tests/components/recorder/test_init.py

```diff
@@ -2071,3 +2071,23 @@ async def test_excluding_attributes_by_integration(
     expected = _state_with_context(hass, entity_id)
     expected.attributes = {"test_attr": 5}
     assert state.as_dict() == expected.as_dict()
+
+
+async def test_lru_increases_with_many_entities(
+    recorder_mock: Recorder, hass: HomeAssistant
+) -> None:
+    """Test that the recorder's internal LRU cache increases with many entities."""
+    # We do not actually want to record 4096 entities, so we mock the recorder
+    # to not actually record anything.
+    with patch(
+        "homeassistant.components.recorder.Recorder._process_state_changed_event_into_session"
+    ):
+        for entity_idx in range(4096):
```

Comment:
Why not patch `hass.states.async_entity_ids_count()` ?

Reply:
adjusted in 12c9b565e48537e139583038a7087b0dc1dc041f

Reply:
I was overthinking this one.